tests/unittests/Caffe2ImporterTest.cpp

```diff
@@ -1549,3 +1549,197 @@ TEST(caffe2, tensorFillsTest) {
     EXPECT_EQ(tensorInt64FillH.raw(i), (int64_t)i);
   }
 }
+
+/// Test loading SparseLengthsWeightedSumFused8BitRowwise. This is created as a
+/// RowwiseQuantizedSparseLengthsWeightedSumNode. The following inputs/outputs
+/// are used/expected for this test. Note that the DATA input is
+/// rowwise-quantized in the init_net proto; the C2 loader will separate the
+/// data from the scales/offsets of each row and place them in separate
+/// Constants for use by the RowwiseQuantizedSparseLengthsWeightedSumNode.
+///    DATA  =   [[2.0, -0.5, 13]]
+///    WEIGHTS = [3, 1, 0, 0, 0, 0, 2, -0.5]
+///    INDICES = [1, 0, 2, 0, 1, 2, 2, 0]
+///    LENGTHS = [3, 0, 3, 2]
+///    OUTPUT =  [[0.5, 0, 0, 25]]
+TEST(caffe2, SparseLengthsWeightedSumFused8BitRowwise) {
+  ExecutionEngine EE{BackendKind::Interpreter};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string NetDescFilename(
+      "tests/models/caffe2Models/"
+      "fused_rowwise_quantized_sparse_lengths_weighted_sum_predict_net.pbtxt");
+  std::string NetWeightFilename(
+      "tests/models/caffe2Models/"
+      "fused_rowwise_quantized_sparse_lengths_weighted_sum_init_net.pbtxt");
+
+  Placeholder *output, *indices, *lengths;
+  Context ctx;
+
+  TypeRef indicesType = F->getParent()->uniqueType(ElemKind::Int64ITy, {8});
+  TypeRef lengthsType = F->getParent()->uniqueType(ElemKind::Int32ITy, {4});
+
+  // Destroy the loader after the graph is loaded since the following execution
+  // will not depend on anyting from the loader.
+  {
+    Caffe2ModelLoader caffe2LD(NetDescFilename, NetWeightFilename,
+                               {"indices", "lengths"},
+                               {indicesType, lengthsType}, *F);
+
+    indices = llvm::dyn_cast<Placeholder>(
+        EXIT_ON_ERR(caffe2LD.getNodeValueByName("indices")));
+    lengths = llvm::dyn_cast<Placeholder>(
+        EXIT_ON_ERR(caffe2LD.getNodeValueByName("lengths")));
+    output = EXIT_ON_ERR(caffe2LD.getSingleOutput());
+  }
+
+  ASSERT_TRUE(indices);
+  ASSERT_TRUE(lengths);
+
+  ctx.allocate(indices)->getHandle<int64_t>() = {
+      1, 0, 2, 0, 1, 2, 2, 0,
+  };
+  ctx.allocate(lengths)->getHandle<int32_t>() = {
```

Comment:
The format here is funny.

Reply:
Yeah for some reason this is what `clang-format` prefers 😕... If there are more than 5 elements then it all goes into a single line like `indices` above, but this is only 4 elements.